### ISSUE TITLE
CommandPermission.fromString doesn't provide suggestions

### ISSUE THREAD

**Issue by JorelAli** [JorelAli/1.13-Command-API#73]
**CommandAPI version**
v2.2

**Minecraft version**
1.14.4

**Describe the bug**
Using `CommandPermission.fromString()` to provide a permission node doesn't work as intended when using the GroupManager permission system. The text remains red when typing (should provide context and show argument colors such as aqua and light yellow), yet the command _will_ run normally and work. Suggestions also don't work.

**My code**
This code reproduces the behavior:
```java
LinkedHashMap<String, Argument> args = new LinkedHashMap<>();
args.put("player-name", new PlayerArgument());
args.put("message-text", new GreedyStringArgument());

CommandAPI.getInstance().register("test", CommandPermission.fromString("debug.command.sendmessage"), args, (commandSender, objects) -> {
    final Player player = (Player) objects[0];
    final String message = (String) objects[1];
    player.sendMessage(ChatColor.translateAlternateColorCodes('&', message));
});
```

**Expected behavior**
Commands work as you'd expect them to, with suggestions and color highlighting

**Screenshots**
![image](https://user-images.githubusercontent.com/4613171/63701481-ef3e7c80-c81c-11e9-964b-d58d8c31c2bc.png)

**Additional context**
_Originally posted by @portlek in https://github.com/JorelAli/1.13-Command-API/issues/66#issuecomment-516439790_

**Comment by JorelAli** [comment]
## Issue update

So, this issue is much more complicated than I initially thought. Basically, what happens is this:

1. A player joins the server
2. Once they join, Minecraft will automatically evaluate what permissions they have for what commands. For example, if you register a command `mycommand` with permission node `mypermission`, it will check if the player has the permission `mypermission` in order to run `mycommand`.
3. Permission plugins realise that a player has joined the server
4. Permission plugins register the permissions that the player can use (normally involves opening a database or file to find out what they can/cannot use)

So basically, permission plugins register player permissions AFTER they are checked, which means it automatically evaluates to `false`, therefore they don't have permission to "use" the command (hence why it's all in red). However, when you actually RUN the command, it checks the permission right there and then (and evaluates to true if they have the permission), which works.

To "fix" this, I believe the permission plugin needs to have support for _offline player permissions_. This probably involves adding Vault as a dependency. I will report if and when further developments are made.

**Comment by JorelAli** [comment]
No, I can't reschedule the check, since it's handled internally by Vanilla Minecraft, not the CommandAPI.
I've done some testing with Vault and I've managed to get a working solution! I'm currently just polishing it off and running my final tests.

**Comment by JorelAli** [comment]
> The reason I mentioned it is there is a method within Command Dispatcher (within NMS package) that resends the declare commands packet, and you would just send that packet again. 

Interesting... I'll look into it.

**Comment by JorelAli** [comment]
@i509VCB Using merely the `PlayerJoinEvent` (no delay or scheduling), it worked perfectly. I'll use this method going forward instead of Vault.

**Comment by willkroboth** [comment]
I believe this was fixed upstream by GroupManager in this commit: https://github.com/ElgarL/GroupManager/commit/e3214b913a8a671a68ee8546fdb738af3fed0905#diff-844ce057a23754e48daf8557a695b41347f115bfcc3ca4a84f2861f38e202d42R433. GroupManager used to update their permissions during the `PlayerJoinEvent`, but that commit switched them to listen for the `PlayerLoginEvent`. That event happens before Vanilla sends the Declare Commands packet, so the packet is correct the first time. GroupManager is now also calling the [`Player#updateCommands` method](https://github.com/ElgarL/GroupManager/blob/4074b2c2f5009e581ed5b4b3e723a05a401052fc/src/org/anjocaido/groupmanager/permissions/BukkitPermissions.java#L218), exactly the same as the CommandAPI after #471.

Another permissions plugin, LuckPerms, also starts loading and applying data [at the `PlayerLoginEvent`](https://github.com/LuckPerms/LuckPerms/blob/master/bukkit/src/main/java/me/lucko/luckperms/bukkit/listeners/BukkitConnectionListener.java#L156). I removed the code from https://github.com/JorelAli/CommandAPI/commit/e2ccbd47358d19fb666a288d181bb46e01f1ed28 and retested this issue on Spigot 1.20 and Paper 1.15.2 using LuckPerms to manage permissions. I was successfully able to tab-complete the command immediately after joining the server.

So, I think this bug is an upstream issue with permission plugins. If this problem reoccurs with a plugin I didn't mention here, they can probably fix it themselves by loading their data in the `PlayerLoginEvent`. The CommandAPI does not seem to need to fix this issue, so this `PlayerJoinEvent` listener can probably be removed:

https://github.com/JorelAli/CommandAPI/blob/0b51a05b2f5333ea4bd7abc676b63f354e6df128/commandapi-platforms/commandapi-bukkit/commandapi-bukkit-core/src/main/java/dev/jorel/commandapi/CommandAPIBukkit.java#L212-L221
